prow/git/client.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"os"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Clients knows how to create clients for repos
+type Clients interface {
```

Comment:
I must say I find the naming very confusing here. Maybe `RepoClientFactory?`?

Reply:
I was weary of getting too Java-ey ;)

Reply:
A rename of this is the one thing I'd really like to see, other than that we can iterate I guess

Reply:
I'd also like to see a rename `Client` vs `Clients` is hard to distinguish when reading and not the clearest semantically.